File: adventofcode-module/src/test/java/nl/mvdr/adventofcode/adventofcode2024/day17/Part1Test.java
package nl.mvdr.adventofcode.adventofcode2024.day17;

import java.util.stream.Stream;

import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

public class Part1Test extends SolverTest<Part1> {

    public Part1Test() {
        super(Part1.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(
                Arguments.of("4,6,3,5,6,3,5,2,1,0", "example-day17-2024-0.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day02/Part1Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day02

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part1Test: FunctionSolverTest<Long>(::solvePart1) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("1227775554", "example-day02.txt")
            )
        }
    }
}